sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TTLState.scala

```diff
@@ -105,17 +109,21 @@ abstract class SingleKeyTTLStateImpl(
   /**
    * Clears any state which has ttl older than [[ttlExpirationMs]].
    */
-  override def clearExpiredState(): Unit = {
+  override def clearExpiredState(): Long = {
     val iterator = store.iterator(ttlColumnFamilyName)
+    var numValuesExpired = 0L
 
     iterator.takeWhile { kv =>
       val expirationMs = kv.key.getLong(0)
       StateTTL.isExpired(expirationMs, ttlExpirationMs)
     }.foreach { kv =>
       val groupingKey = kv.key.getBinary(1)
-      clearIfExpired(groupingKey)
+      if (clearIfExpired(groupingKey)) {
+        numValuesExpired += 1
+      }
```

Comment:
Discussed offline - but we will keep this as it is currently to avoid passing the metrics object to the TTL related classes. Can refactor once the list/map state ttl changes are merged